From 76d774f53844b7359938732bc7ceaefacf7dbb9f Mon Sep 17 00:00:00 2001
From: Vallish Vaidyeshwara <vallish@amazon.com>
Date: Mon, 12 Feb 2018 22:29:56 +0000
Subject: drivers: introduce AMAZON_DRIVER_UPDATES

This provides a central place to maintain out-of-tree drivers.
Renamed from VENDOR_AMAZON because the name was no longer appropriate.

Signed-off-by: Munehisa Kamata <kamatam@amazon.com>
Reviewed-by: Cristian Gafton <gafton@amazon.com>
Reviewed-by: Guru Anbalagane <guruanb@amazon.com>
CR: https://cr.amazon.com/r/6834563/

Reviewed-by: Eduardo Valentin <eduval@amazon.com>
Reviewed-by: Anchal Agarwal <anchalag@amazon.com>
Signed-off-by: Vallish Vaidyeshwara <vallish@amazon.com>
CR: https://cr.amazon.com/r/8244644/
Signed-off-by: Srivatsa S. Bhat <srivatsa@csail.mit.edu>
---
 drivers/Kconfig         |  2 ++
 drivers/Makefile        |  3 +++
 drivers/amazon/Kconfig  | 15 +++++++++++++++
 drivers/amazon/Makefile |  3 +++
 4 files changed, 23 insertions(+)
 create mode 100644 drivers/amazon/Kconfig
 create mode 100644 drivers/amazon/Makefile

diff --git a/drivers/Kconfig b/drivers/Kconfig
index 1d7af3c2ff27..3c5ebdcea0e8 100644
--- a/drivers/Kconfig
+++ b/drivers/Kconfig
@@ -209,4 +209,6 @@ source "drivers/tee/Kconfig"
 
 source "drivers/mux/Kconfig"
 
+source "drivers/amazon/Kconfig"
+
 endmenu
diff --git a/drivers/Makefile b/drivers/Makefile
index 5f5ccdbad21a..786e6c558881 100644
--- a/drivers/Makefile
+++ b/drivers/Makefile
@@ -183,3 +183,6 @@ obj-$(CONFIG_FPGA)		+= fpga/
 obj-$(CONFIG_FSI)		+= fsi/
 obj-$(CONFIG_TEE)		+= tee/
 obj-$(CONFIG_MULTIPLEXER)	+= mux/
+
+# Amazon driver updates
+obj-$(CONFIG_AMAZON_DRIVER_UPDATES)	+= amazon/
diff --git a/drivers/amazon/Kconfig b/drivers/amazon/Kconfig
new file mode 100644
index 000000000000..10ff3828e250
--- /dev/null
+++ b/drivers/amazon/Kconfig
@@ -0,0 +1,15 @@
+#
+# Amazon driver updates configuration
+#
+
+config AMAZON_DRIVER_UPDATES
+	bool "Amazon Driver Updates"
+	default y
+	depends on PCI || EXPERIMENTAL
+	---help---
+	  Amazon driver updates includes out-of-tree drivers and/or modifeid
+	  versions of the drivers present in the stable kernel tree.
+
+if AMAZON_DRIVER_UPDATES
+
+endif # AMAZON_DRIVER_UPDATES
diff --git a/drivers/amazon/Makefile b/drivers/amazon/Makefile
new file mode 100644
index 000000000000..6b4996dcbe52
--- /dev/null
+++ b/drivers/amazon/Makefile
@@ -0,0 +1,3 @@
+#
+# Amazon Driver Updates
+#
-- 
2.14.4

